### PR TITLE
Ms 3 profile

### PR DESCRIPTION
**Pull request by nagui-mostafa**

### CODE REVIEW

File: Backend/src/main/java/com/example/SkyLine/service/NotificationService.java [hunk 1]
```diff
@@ -1,70 +1,249 @@
...
+              return true;
+        }
+        else {
+           System.out.println("You are not allowed to request ");
+           return false;
+        }
+
+
+    }
+ public void approveRequest(int NotificationId) {
```
Comment (AbdelrahmanWael2) [inline Backend/src/main/java/com/example/SkyLine/service/NotificationService.java:72]: I think you should use @Transactional for approve/reject requests and combine them in a single function so when multiple admins/company handle these requests its consistent

File: Backend/src/main/java/com/example/SkyLine/DTO/NotificationRequestDTO.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
+package com.example.SkyLine.DTO;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
+public class NotificationRequestDTO {
+
```
Comment (AH-HESHAM) [inline Backend/src/main/java/com/example/SkyLine/DTO/NotificationRequestDTO.java:8]: put attributes before their getters and setters and for getters and setters you can use lombook for simplicity

File: Backend/src/main/java/com/example/SkyLine/DTO/NotificationRequestDTO.java
Comment (MohAmin22) [inline Backend/src/main/java/com/example/SkyLine/DTO/NotificationRequestDTO.java:1]: Put the fields above the setters an d getters, You should use @Data from lombok to increase the readability, Use @AllArgsConstructor to reduce the boilerplate code.

File: Backend/src/main/java/com/example/SkyLine/DTO/NotificationRequestDTO.java [hunk 3]
```diff
@@ -61,15 +63,47 @@ public void setMessage(String message) {
     private String message;
 
 
+    public String getDate_requested() {
+        return date_requested;
+    }
+
+    public void setDate_requested(LocalDateTime date_requested) {
+
+        this.date_requested = date_requested.format(formatter);;
```
Comment (MohAmin22) [inline Backend/src/main/java/com/example/SkyLine/DTO/NotificationRequestDTO.java:72]: No need to add twice semi-colon

File: Backend/src/main/java/com/example/SkyLine/DTO/UserPromoteDTO.java
Comment (MohAmin22) [inline Backend/src/main/java/com/example/SkyLine/DTO/UserPromoteDTO.java:1]: Add the fileds in the top of the class

**Review by MohAmin22 — APPROVED** [review]
You need to reorganize your code.

**Pull request merged**